core/src/main/scala/com/github/fit51/reactiveconfig/reloadable/Reloadable.scala

```diff
@@ -0,0 +1,248 @@
+        case Stop(stop) =>
+          mapAndStopF[F, (A, B), C](combinedInit, combinedObs, f.tupled, stop)
+        case restart: Restart[F, (A, B), C] =>
+          mapAndRestartF(combinedInit, combinedObs, restart)
+      }
+    } yield result) handleErrorWith {
+      case excp =>
+        log("Failed to construct init value for Reloadable.combineF", excp) >> F.raiseError(excp)
+    }
+
+  private var canceler: F[Unit] = F.unit
+
+  def stop: F[Unit] =
+    canceler
+
+  private def mapEvalAndSkipErrors[G[_]: TaskLike, A, B](ob: Observable[A], f: A => G[B]): Observable[B] =
+    ob.mapEvalF(f.andThen(Task.from(_).attempt)).collect { case Right(b) => b }
+
+  private def mapAndStopF[G[_]: TaskLike, I, O](
+      init: O,
+      obs: Observable[I],
+      f: I => G[O],
+      stop: O => F[_]
+  ): F[Reloadable[F, O]] = {
+    val nextObservable = mapEvalAndSkipErrors[G, I, O](obs, f)
+      .liftByOperator(new ReleasePrevOperator(init, stop.andThen(Task.from(_).void)))
+    Reloadable.apply(init, nextObservable)
+  }
+
+  private def mapAndRestartF[O, I](
+      init: O,
+      obs: Observable[I],
+      restart: Restart[F, I, O]
+  ): F[Reloadable[F, O]] = {
+    val nextObservable = obs
+      .scanEvalF(init.asRight[O].pure[F]) {
+        case (state, newEl) =>
+          val o = state.fold(identity, identity)
+          restart
+            .restart(newEl, o)
+            .map(_.asRight[O])
+            .handleErrorWith {
+              case e =>
+                log("Failed to restart", e).as(o.asLeft[O])
+            }
+      }
+      .collect { case Right(o) => o }
+    Reloadable.apply(init, nextObservable)
+  }
+
+  private def log(message: String, e: Throwable): F[Unit] =
+    Task.delay(logger.error(message, e)).to[F]
+}
+
+private class ReleasePrevOperator[A](init: A, release: A => Task[Unit]) extends Operator[A, A] with StrictLogging {
+
+  import monix.execution.Ack
+  import monix.execution.Ack.Stop
+
+  override def apply(out: Subscriber[A]): Subscriber[A] =
+    new Subscriber[A] {
+      implicit val scheduler              = out.scheduler
+      @volatile private[this] var prev: A = init
+
+      override def onNext(elem: A): Future[Ack] = {
+        try {
+          val nonUsed = prev
+          prev = elem
+          val result = out.onNext(elem)
+          result.onComplete { _ =>
+            releaseWithLogging(nonUsed)
+          }
+          result
+        } catch {
+          case NonFatal(ex) =>
+            onError(ex)
+            Stop
+        }
+      }
+
+      override def onError(ex: Throwable): Unit = {
+        releaseWithLogging(prev)
+        out.onError(ex)
+      }
+
+      override def onComplete(): Unit = {
+        releaseWithLogging(prev)
+        out.onComplete()
+      }
+
+      private def releaseWithLogging(elem: A): Unit =
+        release(elem).runToFuture.onComplete {
+          case Failure(excp) => logger.error(s"Failed to release $elem", excp)
```

Comment:
In such cases we conventionally name Exception as `e`

Reply:
Done.